### DISCUSSION TITLE
SMTP EHLO response for system email

### DISCUSSION THREAD

**Discussion by tguarriello-well** [nextgenhealthcare/connect#5532] [Q&A]
We have several servers on Mirth 3.7.1. One of them (only one) is failing to deliver alert emails through our Google account. If I try to send an email using the Test button on the server config page, it fails. Setup is identical and all servers use the same whitelisted public IP. When we do a trace, the only difference in the connection conversation is that the failing server is initiating with "EHLO localhost" the others send a fully qualified domain name. Is there a way to configure this? I'm not sure if this is Mirth, Java or the OS. Centos 8.22 / JVM 1.8.0.282

**Comment by pacmano1** [comment]
Interesting, I've not had to worry about EHLO on any smtp senders.   

Do the local hostnames on the working instances match the EHLO value presented to Google?

**Reply by tguarriello-well** [reply]
No - that is the odd thing. I can't figure out where the host names are coming from. The EHLO value doesn't match anything I can find on any of the machines. Might be reverse DNS?

**Reply by pacmano1** [reply]
As for reverse DNS, that is probably not correct on thousands of Mirth instances.

**Reply by jonbartels** [reply]
Mirth Connect uses Apache Commons Mail. It *may* be productive to see if Apache Commons Mail reacts to any OS settings or Java system properties when deciding to use EHLO instead of HELO. 

You can see how Mirth Connect calls Apache Commons Mail in https://github.com/nextgenhealthcare/connect/blob/0974c6ad4d6fc683d544f6f23da0def551812ecc/server/src/com/mirth/connect/connectors/smtp/SmtpDispatcher.java

I do not reccomend changing MC code to fix this problem, rather if you can see how MC calls the mail library you can then read the right documentation about that library to learn what settings to change to fix the problem.

**Reply by ab-20-m** [reply]
The property that controls the domain name in the EHLO/HELO command is `mail.smtp.localhost`. The details from the JavaMail(what Apache Commons Mail is using for the lower level bits) is in the `com.sun.smtp` [package summary](https://javaee.github.io/javamail/docs/api/com/sun/mail/smtp/package-summary.html).

> Local host name used in the SMTP HELO or EHLO command. Defaults to InetAddress.getLocalHost().getHostName(). Should not normally need to be set if your JDK and your name service are configured properly.

**Reply by jonbartels** [reply]
What are the contents of `/etc/hostname` on your working and not working instances @tguarriello-well ?

**Comment by pacmano1** [comment]
You got a transparent proxy in there somewhere?
